### PR TITLE
Fraud flag on person

### PR DESCRIPTION
**Pull request by vonElfvin**
Include a green, amber or red flag of person based on fraudulent status of member with same person

Dependency of https://github.com/HedvigInsurance/account-service/pull/51

### CODE REVIEW

File: member-service/src/main/java/com/hedvig/personservice/persons/PersonService.kt [hunk 1]
```diff
@@ -61,18 +63,37 @@ class PersonService @Autowired constructor(
...
-        fun getAllFlags(person: Person): PersonFlags = PersonFlags(
-            debtFlag = DebtService.getDebtFlagByPerson(person)
+        fun calculateAllFlags(person: Person, members: List<MemberEntity>): PersonFlags = PersonFlags(
+            debtFlag = DebtService.calculateDebtFlagByPerson(person),
+            fraudFlag = calculateFraudFlag(members)
         )
 
-        fun getFlag(person: Person): Flag {
-            val personFlags = getAllFlags(person)
+        fun calculateOverallFlag(personFlags: PersonFlags): Flag {
```
Comment (AlexSJ1990) [inline member-service/src/main/java/com/hedvig/personservice/persons/PersonService.kt]: why is this only taking into consideration debtFlag?

Comment (vonElfvin) [reply member-service/src/main/java/com/hedvig/personservice/persons/PersonService.kt]: For now, we only consider a person red based on their UC, we may want to extend it in the future but after discussing with IEX only UC should considered for now.

File: member-service/src/main/java/com/hedvig/personservice/persons/PersonService.kt [hunk 2]
```diff
@@ -61,18 +63,37 @@ class PersonService @Autowired constructor(
...
             return when(personFlags.debtFlag) {
                 Flag.GREEN -> Flag.GREEN
                 Flag.AMBER -> Flag.GREEN
                 Flag.RED -> Flag.RED
```
Comment (palmenhq) [inline member-service/src/main/java/com/hedvig/personservice/persons/PersonService.kt]: what? Why not just `return personflag.debtFlag`?

File: member-service/src/main/java/com/hedvig/personservice/persons/PersonService.kt [hunk 3]
```diff
@@ -61,18 +63,37 @@ class PersonService @Autowired constructor(
...
             }
         }
+
+        fun calculateFraudFlag(members: List<MemberEntity>): Flag {
+            return when (members.maxBy { member -> member.fraudulentStatus }!!.fraudulentStatus) {
```
Comment (palmenhq) [inline member-service/src/main/java/com/hedvig/personservice/persons/PersonService.kt]: what is the ordering of `member.fraudulentStatus`? Feels a but shaky to depend on its order or? Maybe we could do something like

```kt
when {
  members.any { member -> member.fraudulentStatus == FraudulentStatus.CONFIRMED_FRAUD } -> Flag.RED
  members.any { member -> member.fraudulentStatus == FraudulentStatus.SUSPECTED_FRAUD } -> Flag.AMBER
...
}
```

File: member-service/src/main/java/com/hedvig/personservice/persons/PersonService.kt (hunk 2 above)
Comment (vonElfvin) [reply member-service/src/main/java/com/hedvig/personservice/persons/PersonService.kt]: Two reasons really, in this particular calculation, on line 85, we return green flag event though the debt flag is amber. Also in the future we may want to have more logic revolving what a flag is. E.g. include the fraud flag.

Comment (palmenhq) [reply member-service/src/main/java/com/hedvig/personservice/persons/PersonService.kt]: Ah yes of course, didn't catch that haha

**Pull request merged**